demos/keypoints_bounding_boxes/keypoints_bounding_boxes_demo.py

```diff
@@ -0,0 +1,235 @@
+        self.model = yolov5.load(model_path, device=device)
+
+    def __call__(
+        self,
+        img: Union[str, np.ndarray],
+        conf_threshold: float = 0.25,
+        iou_threshold: float = 0.45,
+        image_size: int = 720,
+        classes: Optional[List[int]] = None
+    ) -> torch.tensor:
+
+        self.model.conf = conf_threshold
+        self.model.iou = iou_threshold
+        if classes is not None:
+            self.model.classes = classes
+        detections = self.model(img, size=image_size)
+        return detections
+
+def yolo_detections_to_norfair_detections(
+    yolo_detections: torch.tensor,
+    track_points: str = 'centroid'  # bbox or centroid
+) -> List[Detection]:
+    """convert detections_as_xywh to norfair detections
+    """
+    norfair_detections: List[Detection] = []
+
+    if track_points == 'centroid':
+        detections_as_xywh = yolo_detections.xywh[0]
+        for detection_as_xywh in detections_as_xywh:
+            centroid = np.array(
+                [
+                    detection_as_xywh[0].item(),
+                    detection_as_xywh[1].item()
+                ]
+            )
+            scores = np.array([detection_as_xywh[4].item()])
+            label = int(detection_as_xywh[5].item())
+            norfair_detections.append(
+                Detection(points=centroid, scores=scores, label=label)
+            )
+    elif track_points == 'bbox':
+        detections_as_xyxy = yolo_detections.xyxy[0]
+        for detection_as_xyxy in detections_as_xyxy:
+            bbox = np.array(
+                [
+                    [detection_as_xyxy[0].item(), detection_as_xyxy[1].item()],
+                    [detection_as_xyxy[2].item(), detection_as_xyxy[3].item()]
+                ]
+            )
+            scores = np.array([detection_as_xyxy[4].item(), detection_as_xyxy[4].item()])
+            label = int(detection_as_xyxy[5].item())            
+            norfair_detections.append(
+                Detection(points=bbox, scores=scores, label=label)
+            )
+
+    return norfair_detections
+
+########################################
+
+
+
+
+# Distance function
+def keypoints_distance(detected_pose, tracked_pose):
+    # Use different distances for bounding boxes and keypoints
+    if detected_pose.label != 0:
+        detection_centroid = np.sum(detected_pose.points, axis=0)/len(detected_pose.points)
+        tracked_centroid = np.sum(tracked_pose.estimate, axis=0)/len(detected_pose.points)
+        distances = np.linalg.norm(detection_centroid - tracked_centroid, axis=0)
+        return distances / (KEYPOINT_DIST_THRESHOLD + distances)
+
+    distances = np.linalg.norm(detected_pose.points - tracked_pose.estimate, axis=1)
+    match_num = np.count_nonzero(
+        (distances < KEYPOINT_DIST_THRESHOLD)
+        * (detected_pose.scores > DETECTION_THRESHOLD)
+        * (tracked_pose.last_detection.scores > DETECTION_THRESHOLD)
+    )
+    return 1 / (1 + match_num)
```

Comment:
Would be slightly more readable under an `else` statement, as we are choosing between 2 distinct options.